reference/src/glossary.md

```diff
@@ -129,10 +129,44 @@ niches. For example, the "all bits uninitialized" is an invalid bit-pattern for
 `&mut T`, but this bit-pattern cannot be used by layout optimizations, and is not a
 niche.
 
+#### Place
+
+A *place* (called "lvalue" in C and "glvalue" in C++) is the result of computing a [*place expression*][place-value-expr].
+A place is basically a pointer (pointing to some location in memory, potentially carrying [provenance](#pointer-provenance)), but might contain more information such as size or alignment (the details will have to be determined as the Rust Abstract Machine gets specified more precisely).
+A place has a type.
+Places cannot be "stored" in memory, only [values](#value) can.
+
+The key operations on a place are:
+* storing a [value](#value) of the same type in it (when it is used on the left-hand side of an assignment),
+* turning it into a [pointer value](#value) (when it is used inside `&expr`),
+* and loading a [value](#value) of the same type from it (through the place-to-value coercion).
+
+
+#### Value
+
+A *value* (called "value of the expression" or "rvalue" in C and "prvalue" in C++) is what gets stored in a [place](#place), and also the result of computing a [*value expression*][place-value-expr].
+A value has a type, and it denotes the abstract mathematical concept that is represented by data in our programs.
+For example, a value of type `u8` is a mathematical integer in the range `0..256`.
+Values can be (according to their type) turned into a list of bytes, which is called a [representation](#representation) of the value.
+Values are ephemeral; they arise during the computation of an instruction but are only ever persisted in memory through their representation.
+(This is comparable to how run-time data in a program is ephemeral and is only ever persisted in serialized form.)
+
+#### Representation (relation)
+
+A *representation* of a [value](#value) is a list of bytes that is used to store or "represent" that value in memory.
```

Comment:
By "byte" you probably mean here something more than just 8-bits. 

The other PR defines that term, so it does not need to be done here, but maybe add a (TODO: define byte) here somewhere that gets fixed at some point ?

Reply:
I think this definition makes sense without precisely defining what a "byte" is -- even if the reader thinks "byte == `u8`", there is useful information here. And precisely defining `Byte` will probably be a non-trivial achievement as part of a language spec effort.

Reply:
Maybe just add `byte` to the TODO list below ?

With this PR which mentions provenance and the PR defining provenance "byte" makes one wonder where is provenance in the representation.

Reply:
I don't like that TODO list, IMO issues are better. I will open an issue.